### DISCUSSION TITLE
How can i denote the key columns of jsonl input?

### DISCUSSION THREAD

**Discussion by pbsds** [saulpw/visidata#1531] [Q&A] [answered]
I do not want to press `!` to mark my keys every time i open my generated jsonl data. Is it possible with some input flag to set the key?

**Comment by frosencrantz** [comment, marked as answer]
Hi @pbsds,

There currently is no input flag for this.  There is an ability to run a playback file, and that might be something you want to try.  I think the other solution might be to write a custom loader that uses the regular loader, and does the extra step to make a custom column(s) as key columns.

**Reply by pbsds** [reply]
Too bad, perhaps the jsonl format could be expanded to first accept a meta/header row?

**Reply by frosencrantz** [reply]
@pbsds If you are happy to update your data you could use `vds` format, which is `jsonl` with some leading meta-data about the columns.  And this might be what you want.

You can try saving your `jsonl` data as `vds`.  Not sure if there is a good reference other than the code.  There was some discussion here: https://github.com/saulpw/visidata/discussions/886#discussioncomment-326073

**Reply by pbsds** [reply]
Do you have any good documentation of the format? I've only found [this](https://github.com/saulpw/visidata/blob/develop/sample_data/sample.vds) example, which i can't quite make sense of

**Reply by pbsds** [reply]
I ended up handcrafting a replay:
```
sheet	col	row	longname	input	keystrokes	comment
			open-file	-	o
-	uid		key-col		!	toggle current column as a key column
-	epochs		key-col		!	toggle current column as a key column
-	hp_nonlinearity		key-col		!	toggle current column as a key column
-	val_uloss_intersections		key-col		!	toggle current column as a key column
```
